impexp-core/src/main/java/org/citydb/core/operation/importer/concurrent/DBImportWorker.java

```diff
@@ -190,56 +193,71 @@ private void doWork(CityGML work) {
 		runLock.lock();
 
 		try {
-			if (!shouldWork)
+			if (!shouldWork) {
 				return;
+			}
 
-			long id;
-
-			if (work instanceof Appearance) {
-				// global appearances
-				Appearance appearance = (Appearance)work;
-				id = importer.importGlobalAppearance(appearance);
-
-				if (id != 0)
-					globalAppearanceCounter++;
-			} 
+			if (work instanceof AbstractFeature) {
+				AbstractFeature feature = (AbstractFeature) work;
+
+				// invoke import plugins
+				if (!plugins.isEmpty()) {
+					for (FeatureImportExtension plugin : plugins) {
+						try {
+							feature = plugin.process(feature);
+							if (feature == null) {
+								return;
+							}
+						} catch (PluginException e) {
+							throw new CityGMLImportException("Import plugin " + plugin.getClass().getName() + " threw an exception.", e);
+						}
+					}
+				}
 
-			else if (work instanceof AbstractFeature) {
-				AbstractFeature feature = (AbstractFeature)work;
+				long id;
+				if (work instanceof Appearance) {
```

Comment:
I think we should use the variable `feature` instead of `work`, because the `feature` reflects the latest state for the CityGML import

Reply:
Yes, changed with 1b4e514

Reply:
Nice